### PR TITLE
feat: added RawOutputDataConfig in ExecutionSpec #patch

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>
# TL;DR
added RawOutputDataConfig in ExecutionSpec, so we can override LaunchPlan's RawOutputDataConfig at CreateExecutionRequest time

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
image

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2230

## Follow-up issue
_NA_

### CODE REVIEW

**Review by anrusina — COMMENTED** [review]
Hi Kevin, 
Can you please rebase on top of latest master. Today a big eslint/prettier changes were merged, which already took care of all spaces changes. 
Unfortunately your PR was done before that - and as result, it is hard to follow which exactly changes were added by you, and which are just styling update.

Please, let me know if it's a hustle, and I will try to update it from my side (looks like you have your own fork of flyteconsole, which makes it a bit more complicated..).

**Comment by pingsutw** [comment]
@anrusina I just rebased it, but the tests failed. Do you know how to fix it?

File: package.json [hunk 1]
```diff
@@ -82,7 +82,7 @@
         "@commitlint/cli": "^8.3.5",
         "@commitlint/config-conventional": "^8.3.4",
         "@date-io/moment": "1.3.9",
-        "@flyteorg/flyteidl": "0.21.24",
+        "@flyteorg/flyteidl": "0.23.1",
```
Comment (anrusina) [inline package.json:85]: Did smoke test, the new field looks good.
Is it expected, that if provided Config is not accessible it is using default value? 
See here:
https://user-images.githubusercontent.com/55718143/158853593-7c554fba-b544-4017-9607-41fe7ddc1c4a.mp4
Otherwise LGTM, and I can merge it later.

**Review by anrusina — APPROVED** [review]
Lgtm

File: package.json (hunk 1 above)
Comment (pingsutw) [reply package.json:85]: yes, We should use default value if bucket is not accessible
